src/changeset.rs

```diff
@@ -0,0 +1,192 @@
+use crate::repo::{ChangeError, Insertable, TableRef, Validatable};
+use sea_query::{Alias, Iden, PostgresQueryBuilder, Query, SimpleExpr};
+use std::mem;
+use std::vec::Vec;
+
+pub trait Valuable {
+    fn value(&self) -> SimpleExpr;
+}
+
+macro_rules! panic_if_not_match {
+    (
+        <$enum:ty>::$variant:ident($inner_val:ident), $value:ident
+    ) => {{
+        type Enum = $enum;
+        if let Enum::$variant($inner_val) = $value {
+            $inner_val
+        } else {
+            panic!("Pattern does not match")
+        }
+    }};
+}
+
+// TODO see if you can find a way to keep the closures off the heap
```

Comment:
I played around with this some and it doesn't really seem feasible. I turned the vec into an array slice but then it didn't like it when I put the closures in boxes. Said I had a mismatched type since it was expecting Fn(T) but was getting closures... so... I guess this is fine. The UX is decent at least.